actors/eam/src/lib.rs

```diff
@@ -165,19 +148,10 @@ where
 {
     let caller_id = rt.message().caller().id().unwrap();
 
-    let addr = rt.lookup_address(caller_id);
-
-    match addr.map(|a| *a.payload()) {
-        Some(Payload::Delegated(eth)) => {
-            if eth.namespace() == EAM_ACTOR_ID {
-                Ok(EthAddress(eth.subaddress().try_into().unwrap()))
-            } else {
-                Err(ActorError::assertion_failed("Caller is not in the EVM namespace.".into()))
-            }
-        }
-        _ => Err(ActorError::assertion_failed(
-            "All FEVM actors should have a delegated address.".to_string(),
-        )),
+    let addr = *rt.lookup_address(caller_id).unwrap().payload();
+    match addr {
```

Comment:
nit: I'd actually _not_ dereference (save rust some copying until it actually needs to).

Reply:
well it does need to copy, since it returns a refrence to a value that is dropped before the next line

Reply:
eh or something else, just better ordering